src/otx/core/data/transform_libs/mmaction.py

```diff
@@ -37,7 +46,162 @@ def __call__(self, entity: ActionClsDataEntity) -> dict:
 
 
 @TRANSFORMS.register_module(force=True)
-class PackActionInputs(MMPackActionInputs):
+class LoadVideoForDetection:
+    """Class to convert OTXDataEntity to dict for MMAction framework."""
+
+    fps: int = 30
+    timestamp_start: int = 900
+
+    def __call__(self, entity: ActionDetDataEntity) -> dict:
+        """Transform ActionClsDataEntity to MMAction data dictionary format."""
+        results: dict[str, Any] = {}
+        results["modality"] = "RGB"
+        results["fps"] = self.fps
+        results["timestamp_start"] = self.timestamp_start
+        results["filename_tmpl"] = "{video}_{idx:04d}.{ext}"
+        results["proposal_file"] = entity.proposal_file
+        results["ori_shape"] = entity.img_info.ori_shape
+
+        if entity.frame_path is not None:
+            frame_dir, extension, shot_info, timestamp = self._get_meta_info(entity.frame_path)
+            results["timestamp"] = timestamp
+            results["shot_info"] = shot_info
+            results["frame_dir"] = frame_dir
+            results["extension"] = extension
+
+        results["__otx__"] = entity
+
+        return results
+
+    @staticmethod
+    def _get_meta_info(frame_path: str) -> tuple[Path, str, tuple[int, int], int]:
+        frame_dir = Path(frame_path).parent
+        extension = Path(frame_path).suffix[1:]
+        shot_info = (1, len(os.listdir(frame_dir)))
+        timestamp = int(Path(frame_path).stem.split("_")[-1])
+
+        return frame_dir, extension, shot_info, timestamp
+
+
+@TRANSFORMS.register_module(force=True)
+class LoadAnnotations:
```

Comment:
If this class overrides the existing one from mmaction, please notify it explicitly such as

```suggestion
@TRANSFORMS.register_module(force=True)
class LoadAnnotations(_LoadAnnotations):
```

Reply:
[d14d790](https://github.com/openvinotoolkit/training_extensions/pull/2747/commits/d14d7905ba067e75b84a1441e2bac302ab299ae8) LoadAnnotation is a new function in OTX, so I removed force=True parameters

Reply:
Then, why do we need this? How can `mmaction` originally import annotations? It is not desired to register a new own transform. This is because we mandatorily require an user who wants to migrate `mmaction` based workflow to OTX 2.0 to include that transform. We want to avoid it as much as possible.

Reply:
mmaction adds gt_bboxes and gt_labels in Dataset class. (https://github.com/open-mmlab/mmaction2/blob/4d6c93474730cad2f25e51109adcf96824efc7a3/mmaction/datasets/ava_dataset.py#L295)
In OTX we need to add this function since we need to add bboxes and labels info to data dictionary from otx entity